File: binary_search.py
def binary_search(nums, n):
    l = 0
    r = len(nums)
    steps = 0

    while l < r:
        steps+=1
        mid = int((l + r) / 2)
        
        if nums[mid] == n:
            print("steps: ", steps)
            return mid
        elif nums[mid] < n:
            l = mid + 1
        else:
            r = mid
    return -1

a = lista = list(range(1, 6))
b = lista = list(range(1, 11))
c = lista = list(range(1, 41))

# binary_search(a, 3)
binary_search(b, 7)
# binary_search(c, 3)

    
        


        
                    

            
            

    
    
    
    

File: java/BinarySearch.java
package java;
public class BinarySearch {
    public static void main(String[] args) {
        System.out.println(Solution.search(new int[] {1, 2, 3, 4, 5}, 3));
    }
}

class Solution {
    public static int search(int[] nums, int target) {
        int left = 0, right = nums.length - 1, steps = 0;

        while (left <= right) {
            steps++;
            int mid = (left + right) / 2;
            
            if(nums[mid] == target){
                System.out.println("Steps: " + steps);
                return nums[mid];
            } else if(target < nums[mid]){
                right = mid - 1;
            } else {
                left = mid + 1; 
            }
        }

        return -1;
    }
}